### DISCUSSION TITLE
TypeError: Cannot read property 'Tag' of undefined

### DISCUSSION THREAD

**Discussion by dribeirodev** [csandman/chakra-react-select#108] [Q&A] [answered]
I added your package to use multi select and my tests broke. Do you have any idea what it might be?

![image](https://user-images.githubusercontent.com/18536415/144134312-8df20c8e-5ff4-4abc-ac56-acd60eee8b9f.png)

![image](https://user-images.githubusercontent.com/18536415/144134431-3dd7e8cb-a856-43b7-985a-bd992fa3aa86.png)

**Comment by csandman** [comment]
I need a little more info to have any idea whats going on with this?

- What version of `chakra-react-select` are you on?
- What testing library are you using?
- How are you implementing `chakra-react-select`? (props passed into it, parent element, etc)
- How is your test interacting with `chakra-react-select` or sibling elements?

**Comment by keithburgie-rts** [comment]
I'm also getting this error all of a sudden. The component was working fine and now it's not.

![Screen Shot 2021-12-07 at 10 49 56 PM](https://user-images.githubusercontent.com/58005205/145145206-7bce7c64-55a2-4cab-a13f-091fbd8f46c8.png)

**Comment by jonahallibone** [comment]
@keeeeith are you on the latest version of the component?

**Comment by keithburgie-rts** [comment]
Yeah I think it was related to a dependency duplicate/conflict of the Emotion package. While searching I came across others who'd had a similar dependency problem with react-select. When I got that straightened out the error went away.

**Comment by csandman** [comment]
@keeeeith What did you end up having to do to get the conflict sorted out?

**Comment by keithburgie-rts** [comment]
@csandman The error started before my planned changes, but the changes were a pretty significant overhaul of my project so I don't think details will be much help to anyone else. I had a locally imported and symlinked component library that had Chakra UI as a dependency and I think that was part of the Emotion conflict.

Looking closer at my own screenshot right now, maybe my chakraTheme.components object was over-riding this package? I also import theme as chakraTheme and have a `components` object that doesn't include `Tag`.
```javascript
import { extendTheme, theme as chakraTheme } from "@chakra-ui/react"
```

**Comment by magnuslinden** [comment, marked as answer]
I'm seeing the same `TypeError: Cannot read properties of undefined (reading 'Tag')` error in my tests, running:

```json
{
  "@chakra-ui/react": "^1.7.2",
  "react-select": "^4.3.1",
  "chakra-react-select": "^1.3.4",
  "@emotion/react": "^11.7.0",
}
```

and testing with jest + react testing library.

**EDIT:**

To fix this I simply had to wrap my tests with `<ChakraProvider>[...]</ChakraProvider>`. Figured that out by looking at Chakra's own testing utilities found here: https://github.com/chakra-ui/chakra-ui/blob/main/tooling/test-utils/src/test-utils.tsx

I've got a bunch of tests that run fine without being wrapped in `<ChakraProvider>`, which threw me off when tests that had this package included failed, but not other ChakraUI components.

**Comment by csandman** [comment]
@magnuslinden That would probably be the standard answer to this question. This component accesses the global Chakra theme for styles in case a user has overridden any of them. I tried to avoid hardcoding any chakra styles where I could to keep things looking consistent.

@dribeirodev Does wrapping your tests in a `<ChakraProvider />` fix your problem?
